### PR TITLE
Make `ActionGroup` compatible with existing Parameter conventions

### PR DESCRIPTION
**Pull request by ntjess**
- The name didn't end with `Parameter` which is unlike every other class
- `sigActivated` should emit `self` instead of nothing
- `emitStateChanged` should be fired during `activate`

### CODE REVIEW

**Review by j9ac9k — APPROVED** [review]
LGTM!

**Comment by ksunden** [comment]
Any public API concerns from the name change?

**Comment by ntjess** [comment]
@ksunden This class was introduced in 0.13, so thankfully the API issue isn't binding at this point. Also, the parameter type is considered "private" (leading underscore), so even since the 0.13 release I don't think it would be widely adopted: https://github.com/pyqtgraph/pyqtgraph/blob/d6f2da977ece7cebca7e09e30b1b3e32a5f9762b/pyqtgraph/parametertree/parameterTypes/__init__.py#L33

**Comment by j9ac9k** [comment]
I think we should probably leave the old class with a deprecation warning, saying it will be removed in the first pyqtgraph release after January, 2023.  The old class, calling the "correct" named class.  If it wasn't already in a release I would say we can just rename it, but it's been like this for long enough it _might_ impact someone.

**Comment by ntjess** [comment]
Very understandable. Will do

File: pyqtgraph/parametertree/parameterTypes/actiongroup.py [hunk 1]
```diff
@@ -58,3 +60,19 @@ def setButtonOpts(self, **opts):
...
+
+class ActionGroup(ActionGroupParameter):
+    sigActivated = QtCore.Signal()
+
+    def __init__(self, **opts):
+        warnings.warn(
+            "`ActionGroup` is deprecated and will be removed in the first release after "
+            "January 2023. Use `ActionGroupParameter` instead. See "
+            "https://github.com/pyqtgraph/pyqtgraph/pull/2505 for details.",
+            DeprecationWarning,
```
Comment (j9ac9k) [inline pyqtgraph/parametertree/parameterTypes/actiongroup.py:73]: probably want to add `stacklevel=2` after `DeprecationWarning`; see any of our usage of deprecation warnings in the library 😆 (it's an awful default in the standard lib)

**Comment by ntjess** [comment]
Good catch!

**Comment by ntjess** [comment]
@j9ac9k Ok to merge?

**Comment by j9ac9k** [comment]
whops, yes! Sorry! I'll press the buttan!

**Pull request merged**